examples/vae/net.py

```diff
@@ -1,67 +1,100 @@
-import six
+import numpy as np
 
 import chainer
+from chainer import cuda
+import chainer.distributions as D
 import chainer.functions as F
-from chainer.functions.loss.vae import gaussian_kl_divergence
 import chainer.links as L
+from chainer import reporter
 
 
-class VAE(chainer.Chain):
-    """Variational AutoEncoder"""
+class AvgELBOLoss(chainer.Chain):
+    def __init__(self, encoder, decoder, prior):
+        super(AvgELBOLoss, self).__init__()
+
+        with self.init_scope():
+            self.encoder = encoder
+            self.decoder = decoder
+            self.prior = prior
+
+        self.loss = None
+        self.rec = None
+        self.penalty = None
+
+    def __call__(self, x):
+        q_z = self.encoder(x)
+        z = q_z.sample()
+        p_x = self.decoder(z)
+        p_z = self.prior()
+
+        self.loss = None
+        self.rec = None
+        self.penalty = None
+        self.rec = F.mean(F.sum(p_x.log_prob(x), axis=-1))
+        self.penalty = F.mean(
```

Comment:
I think it is better to use `kl_divergence`.
https://docs.chainer.org/en/latest/reference/generated/chainer.kl_divergence.html#chainer.kl_divergence

Reply:
I'll change it, thank you.